org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/lsp/ExecuteCommandProposedClient.java

```diff
@@ -0,0 +1,23 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.lsp;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.lsp4j.ExecuteCommandParams;
+import org.eclipse.lsp4j.jsonrpc.services.JsonRequest;
+
+public interface ExecuteCommandProposedClient {
+
+	@JsonRequest("workspace/executeCommand")
```

Comment:
there's already a "workspace/executeCommand" command on the [server side](https://github.com/eclipse/eclipse.jdt.ls/blob/1a6cb087019473d7a7bdc70f71f62c223b41203a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java#L372), so I suggest using a different name

Reply:
I thought it logical to use the same name as it actually does exactly the same thing just the message are flowing in the opposite direction. But sure... could change it to 'executeClientCommand' ot maybe 'client/executeCommand'? What is more logical?

Reply:
either one is fine I guess, I have no strong opinion for one or the other